File: frontend/src/app/registration/registration.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
import { catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import { AuthService } from '../login/auth.service';

@Injectable({
  providedIn: 'root'
})
export class RegistrationService {

  constructor(private _http: HttpClient, private authService: AuthService) { }

  registerPatient(patient: any) {
    return this._http.post<any>("http://localhost:8080/registration/registerPatient", patient)
               .pipe(catchError(this.errorHander));
  }

  registerSystemAdministrator(systemAdministrator: any) {
    return this._http.post<any>("http://localhost:8080/registration/registerSystemAdministrator", systemAdministrator)
               .pipe(catchError(this.errorHander));
  }

  registerPharmacyAdministrator(pharmacyAdministrator: any) {
    return this._http.post<any>("http://localhost:8080/registration/registerPharmacyAdministrator", pharmacyAdministrator)
               .pipe(catchError(this.errorHander));
  }

  registerDermatologist(dermatologist: any) {
    return this._http.post<any>("http://localhost:8080/registration/registerDermatologist", dermatologist)
               .pipe(catchError(this.errorHander));
  }

  registerSupplier(supplier: any) {
    return this._http.post<any>("http://localhost:8080/registration/registerSupplier", supplier)
               .pipe(catchError(this.errorHander));
  }

  errorHander(error: HttpErrorResponse) {
    return throwError(error);
  }
}

File: backend/src/main/java/com/MRSISA2021_T15/service/RegistrationService.java
package com.MRSISA2021_T15.service;

import org.springframework.web.servlet.ModelAndView;

import com.MRSISA2021_T15.model.Dermatologist;
import com.MRSISA2021_T15.model.Patient;
import com.MRSISA2021_T15.model.PharmacyAdmin;
import com.MRSISA2021_T15.model.Supplier;
import com.MRSISA2021_T15.model.SystemAdmin;

public interface RegistrationService {
	
	String registerPatient(Patient patient);
	
	ModelAndView confirmAccount(ModelAndView modelAndView, String confirmationToken);
	
	String registerSystemAdmin(SystemAdmin systemAdmin);
	
	String registerDermatologist(Dermatologist dermatologist);
	
	String registerSupplier(Supplier supplier);
	
	String registerPharmacyAdministrator(PharmacyAdmin pharmacyAdmin);

}